cpp/ql/lib/semmle/code/cpp/dataflow/internal/FlowSummaryImpl.qll

```diff
@@ -39,7 +39,8 @@ module Input implements InputSig<DataFlowImplSpecific::CppDataFlow> {
   string encodeContent(ContentSet cs, string arg) {
     exists(FieldContent c |
       cs.isSingleton(c) and
-      result = indirectionString(c.getIndirectionIndex()) + "Field" and
+      // FieldContent indices have 0 for the address, 1 for content, so we need to subtract one.
+      result = indirectionString(c.getIndirectionIndex() - 1) + "Field" and
```

Comment:
I think using a `toString` in query logic would be opening a can of worms, even if it appears safe in this case.  I can implement a `getName` style predicate and use that if you'd prefer that the `- 1` logic isn't duplicated?

Reply:
Since the change from `*Field` to `Field[*]` syntax the new predicate would only be used in one place now (previously it would have been used in two places I think).  So I'm not going to do this change now.